### PR TITLE
Added headersToDownstreamOnAllow example in ext authz task page

### LINKED ISSUE #12764: Add headersToDownstreamOnAllow example in ext authz task page
Add `headersToDownstreamOnAllow ` to https://istio.io/latest/docs/tasks/security/authorization/authz-custom/#define-the-external-authorizer. Which is described in https://github.com/istio/istio/issues/36950 and implemented in https://github.com/istio/istio/pull/37677

### PR DESCRIPTION
**Pull request by SuperAayush**
Added headersToDownstreamOnAllow to https://istio.io/latest/docs/tasks/security/authorization/authz-custom/#define-the-external-authorizer.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [X] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

Closes #12764

### CODE REVIEW

**Comment by hanxiaop** [comment]
Looks like it needs to be edited in `snips.sh` and run `make gen` to generate the page.

**Comment by ericvn** [comment]
@SuperAayush You will need to satisfy the CLA bot

Also a note to @istio/wg-security-maintainers to review as this page is owned by them.

**Comment by SuperAayush** [comment]
> @SuperAayush You will need to satisfy the CLA bot
> 
> Also a note to @istio/wg-security-maintainers to review as this page is owned by them.

Hey @ericvn!!
Did not do because of this: https://istio.slack.com/archives/CFXS256EQ/p1677255044214359

Do I still need to go for google cla?

**Comment by ericvn** [comment]
@SuperAayush For now for this PR to merge, the Google CLA is required as can been seen in the checks. I need to chat with @therealmitchconnors or @craigbox to see why the EasyCLA is not in this repo yet (it seems to be in the others). And then if the check is an either/or at this point.

**Comment by ericvn** [comment]
It may be that this PR was created before the EasyCLA stuff was added to the repo, since looking at a more recent PR in this repo, I do see it.

**Comment by SuperAayush** [comment]
> It may be that this PR was created before the EasyCLA stuff was added to the repo, since looking at a more recent PR in this repo, I do see it.

Done ✅.

File: content/en/docs/tasks/security/authorization/authz-custom/index.md [hunk 1]
```diff
@@ -143,6 +143,7 @@ allows requests with the header `x-ext-authz: allow`.
             port: "4180" # The default port used by oauth2-proxy.
             includeRequestHeadersInCheck: ["authorization", "cookie"] # headers sent to the oauth2-proxy in the check request.
             headersToUpstreamOnAllow: ["authorization", "path", "x-auth-request-user", "x-auth-request-email", "x-auth-request-access-token"] # headers sent to backend application when request is allowed.
+            headersToDownstreamOnAllow:["content-type", "set-cookie"] # headers sent back to the client when request is allowed.
```
Comment (lei-tang) [inline content/en/docs/tasks/security/authorization/authz-custom/index.md:146]: Why this line needs to be added?

Comment (hanxiaop) [reply content/en/docs/tasks/security/authorization/authz-custom/index.md:146]: This is to inform users that they can also set this option if needed, which already has been implemented.

**Comment by hanxiaop** [comment]
https://deploy-preview-12791--preliminary-istio.netlify.app/latest/docs/tasks/security/authorization/authz-custom/#define-the-external-authorizer looks like there's still some format issue? The font color is not the same as others.

**Comment by craigbox** [comment]
There may need to be a space between `headersToDownstreamOnAllow:` and `[` ?

**Comment by SuperAayush** [comment]
> There may need to be a space between `headersToDownstreamOnAllow:` and `[` ?

Should I change it and commit again?

File: content/en/docs/tasks/security/authorization/authz-custom/index.md (hunk 1 above)
Comment (ericvn) [inline content/en/docs/tasks/security/authorization/authz-custom/index.md:146]:
```suggestion
            headersToDownstreamOnAllow: ["content-type", "set-cookie"] # headers sent back to the client when request is allowed.
```
as Craig noted.

File: content/en/docs/tasks/security/authorization/authz-custom/snips.sh [hunk 2]
```diff
@@ -100,6 +100,7 @@ data:
         port: "4180" # The default port used by oauth2-proxy.
         includeRequestHeadersInCheck: ["authorization", "cookie"] # headers sent to the oauth2-proxy in the check request.
         headersToUpstreamOnAllow: ["authorization", "path", "x-auth-request-user", "x-auth-request-email", "x-auth-request-access-token"] # headers sent to backend application when request is allowed.
+        headersToDownstreamOnAllow:["content-type", "set-cookie"] # headers sent back to the client when request is allowed.
```
Comment (ericvn) [inline content/en/docs/tasks/security/authorization/authz-custom/snips.sh:103]: `make scripts` will fix this one.

**Review by hanxiaop — APPROVED** [review]

**Review by lei-tang — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Pull request merged**